### PR TITLE
KRACOEUS-8728

### PR DESCRIPTION
**Pull request by rmancher**
KRACOEUS-8728
Fix to open budget in view mode when proposal is in view mode
Also in account return back to proposal.

### CODE REVIEW

File: coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetCommonController.java [hunk 1]
```diff
@@ -68,7 +68,8 @@ public ModelAndView defaultMapping(@ModelAttribute("KualiForm") ProposalBudgetFo
 
 	@MethodAccessible
 	@Transactional @RequestMapping(params="methodToCall=start")
-	public ModelAndView start(@RequestParam("budgetId") Long budgetId, @RequestParam("auditActivated") String auditActivated, @ModelAttribute("KualiForm") ProposalBudgetForm form) {
+	public ModelAndView start(@RequestParam("budgetId") Long budgetId, @RequestParam("viewOnly") String viewOnly, @RequestParam("auditActivated") String auditActivated, @ModelAttribute("KualiForm") ProposalBudgetForm form) {
+		boolean inViewMode = Boolean.parseBoolean(viewOnly);
 		form.setBudget(loadBudget(budgetId));
         getProposalBudgetLockService().establishBudgetLock(form.getBudget());
```
Comment (blackcathacker) [inline coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetCommonController.java]: So view mode should affect the locking. Looks like we need more work on locking and readonly mode?

File: coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetForm.java [hunk 2]
```diff
@@ -202,4 +203,12 @@ public void setDataValidationItems(List<DataValidationItem> dataValidationItems)
 		this.dataValidationItems = dataValidationItems;
 	}
 
+	public boolean isViewOnly() {
+		return viewOnly;
+	}
+
+	public void setViewOnly(boolean viewOnly) {
+		this.viewOnly = viewOnly;
+	}
```
Comment (blackcathacker) [inline coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetForm.java]: Do we really need an additional form property for this? Can't we do this using the view.readOnly or just form.canEditView?

File: coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetCommonController.java (hunk 1 above)
Comment (rmancher) [inline coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetCommonController.java]: In readonly/view can we skip locking?

File: coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetForm.java (hunk 2 above)
Comment (rmancher) [inline coeus-impl/src/main/java/org/kuali/coeus/propdev/impl/budget/core/ProposalBudgetForm.java]: Yeah, I tried that initially but I see that budget goes readOnly when it is set as complete.
So thought better not rely on that.

**Pull request merged**